exec/java-exec/src/main/java/org/apache/drill/exec/planner/common/JoinControl.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.drill.exec.planner.common;
+
+/**
+ * For the int type control,
+ * the meaning of each bit start from lowest:
+ * bit 0: intersect or not, 0 -- default(no intersect), 1 -- INTERSECT (DISTINCT as default)
+ * bit 1: intersect type, 0 -- default (DISTINCT), 1 -- INTERSECT_ALL
+ */
+public class JoinControl {
```

Comment:
Looks like the JoinControl currently is not being actually used by the runtime operators; is it just a preparation for future use ?

(A little off the subject)  And it is used to convey information from the planner to the operator. I was recently planning passing other information (e.g., semi-join, anti-semi, broadcast, between lateral and unnest ...) -- any such new bit of information requires so many changes, and extending method signatures; there should be an easier way to do that - like passing the whole plan (but that's another project ....).

Reply:
The JoinControl is actually used in IndexIntersectPlanGenerator, NonCoveringIndexPlanGenerator and a few other places.  You may need to expand those files since GitHub does not automatically load the diffs by default. 

For the broader question of passing around the join type and related information in a more componentized way, yes this has been a pending item that planner needs to provide.  Pls feel free to create a JIRA and if you have any initial thoughts you can start proposing it there.

Reply:
I cherry-picked the 5 commits on top of the (almost) latest master, and searching through the code I could not find who **consumes** the `joinControl` , that is - that value is being passed around in many method calls, and initializes local fields, but no where in the code a **decision** is made based on that value ... 
   Many methods that seem to analyze this value are unused. Like `isIntersect()`, `isIntersectDistinct()`, `isIntersectAll()` in `IndexGroup`,  or `getJoinControl()` in `HashTableConfig`.
    And also unused are the fields `joinControl` in `HashJoinTableTemplate`, or `joinControl` in `HashJoinPrel`.

Reply:
Hmm...it seems a small part of the code in HashJoinBatch may not have been merged correctly .. either when I did the recent merge or when previously we merged the index code with the hash join spilling code.  On a separate private branch, I do see `JoinControl` being used for a decision point.  I will talk to you offline about this. 

I am also tagging @chunhui-shi who is a co-author and had added the `JoinControl` specifically to distinguish between INTERSECT_DISTINCT and INTERSECT_ALL.

Reply:
JoinControl has to be used in HashJoin runtime operator, basically it is to allow the operator to skip some operations when it is index join(so we are expecting INTERSECT and nothing else). I don't have the original code at hand, but I think it should be in the first MapR 6.0 release. Since this part is only about performance and correctness won't be impacted, if it was lost during some merge process people might not notice it.

Reply:
@Ben-Zvi I have incorporated the joinControl logic as part of the hash join probe phase.  Please see changes in [1].  I haven't done it for build phase since it will be superseded by the semi-join changes.  Can you pls review this change ?

[1] https://github.com/apache/drill/pull/1466/commits/5330fd684a20587e9e860d7894e3d3602a6d0495